MapboxVision/Services/VideoSource/ObservableVideoSource.swift

```diff
@@ -6,36 +6,61 @@ import Foundation
  You may inherit your video source from this class to avoid handling observers yourself.
  */
 open class ObservableVideoSource: NSObject, VideoSource {
```

Comment:
using a lock in all the methods brings a possibility of facing deadlock when someone will try to add/remove an observer inside the `notify`'s closure, let's poses this in a doc comment